src/python/pants/vcs/changed_integration_test.py

```diff
@@ -157,6 +157,12 @@ def test_delete_generated_target(repo: str) -> None:
     for dependees in DependeesOption:
         assert_list_stdout(repo, ["//:lib"], dependees=dependees)
```

Comment:
This test is really important. I'm pretty sure our handling of deleted files is wrong, but I wanted to preserve the original semantics for now.

Reply:
Not sure what you mean here... is it something that is worth marking with a TODO?

FWIW: I think that we must report any generator that might have owned a deleted file, because otherwise we won't end up with any changed root at all.

Reply:
https://github.com/pantsbuild/pants/issues/13232